addons/portal/controllers/attachment.py

```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.http import request
+
+from odoo.addons.mail.controllers.attachment import AttachmentController
+from odoo.addons.portal.utils import get_portal_partner
+
+
+class PortalAttachmentController(AttachmentController):
+    def _is_allowed_to_delete(self, message, **kwargs):
+        thread = request.env[message.model].browse(message.res_id)
+        if (
+            thread
```

Comment:
You'd need to check existence of model and res_id beforehand rather than checking existence of thread.